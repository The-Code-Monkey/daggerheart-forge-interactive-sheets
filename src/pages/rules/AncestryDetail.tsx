import { useParams, Link } from "react-router-dom";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { ArrowLeft } from "lucide-react";
import { JSX } from "react";

const AncestryDetail = (): JSX.Element => {
  const { ancestryName } = useParams();

  const ancestries = {};

  const ancestryData =
    ancestries[ancestryName?.toLowerCase() as keyof typeof ancestries];

  if (!ancestryData) {
    return (
      <div className="min-h-screen bg-gradient-to-b from-purple-900 via-purple-800 to-slate-900 py-8 px-4">
        <div className="max-w-4xl mx-auto text-center">
          <h1 className="text-4xl font-bold text-white mb-4">
            Ancestry Not Found
          </h1>
          <Link to="/game-rules">
            <Button className="bg-purple-600 hover:bg-purple-700">
              <ArrowLeft className="w-4 h-4 mr-2" />
              Back to Game Rules
            </Button>
          </Link>
        </div>
      </div>
    );
  }

  return (
    <div className="min-h-screen bg-gradient-to-b from-purple-900 via-purple-800 to-slate-900 py-8 px-4">
      <div className="max-w-4xl mx-auto">
        {/* Header */}
        <div className="mb-8">
          <Link to="/game-rules">
            <Button
              variant="ghost"
              className="text-purple-200 hover:text-white mb-4"
            >
              <ArrowLeft className="w-4 h-4 mr-2" />
              Back to Game Rules
            </Button>
          </Link>

          <div className="mb-4">
            <h1 className="text-4xl font-bold text-white mb-2">
              {ancestryData.name}
            </h1>
            <p className="text-xl text-purple-200">
              {ancestryData.description}
            </p>
          </div>
        </div>

        {/* Ancestry Stats */}
        <div className="grid grid-cols-1 md:grid-cols-2 gap-6 mb-8">
          <Card className="bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-purple-500/30 backdrop-blur-sm">
            <CardHeader>
              <CardTitle className="text-white">Physical Traits</CardTitle>
            </CardHeader>
            <CardContent className="space-y-2">
              <div className="flex justify-between">
                <span className="text-purple-200">Size:</span>
                <span className="text-white">{ancestryData.size}</span>
              </div>
              <div className="flex justify-between">
                <span className="text-purple-200">Speed:</span>
                <span className="text-white">{ancestryData.speed}</span>
              </div>
              <div className="flex justify-between">
                <span className="text-purple-200">Lifespan:</span>
                <span className="text-white">{ancestryData.lifespan}</span>
              </div>
            </CardContent>
          </Card>

          <Card className="bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-purple-500/30 backdrop-blur-sm">
            <CardHeader>
              <CardTitle className="text-white">Languages</CardTitle>
            </CardHeader>
            <CardContent>
              <div className="flex flex-wrap gap-2">
                {ancestryData.languages.map((language, index) => (
                  <Badge
                    key={index}
                    variant="secondary"
                    className="bg-purple-600/50 text-white"
                  >
                    {language}
                  </Badge>
                ))}
              </div>
            </CardContent>
          </Card>
        </div>

        {/* Ancestry Traits */}
        <Card className="bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-purple-500/30 backdrop-blur-sm mb-8">
          <CardHeader>
            <CardTitle className="text-white">Ancestry Traits</CardTitle>
            <CardDescription className="text-purple-200">
              Special abilities that all {ancestryData.name} possess
            </CardDescription>
          </CardHeader>
          <CardContent>
            <div className="space-y-4">
              {ancestryData.traits.map((trait, index) => (
                <div key={index} className="border-l-4 border-yellow-500 pl-4">
                  <h3 className="text-lg font-semibold text-white mb-2">
                    {trait.name}
                  </h3>
                  <p className="text-purple-200">{trait.description}</p>
                </div>
              ))}
            </div>
          </CardContent>
        </Card>

        {/* Culture */}
        <Card className="bg-gradient-to-br from-slate-800/80 to-slate-900/80 border-purple-500/30 backdrop-blur-sm">
          <CardHeader>
            <CardTitle className="text-white">Culture & Society</CardTitle>
          </CardHeader>
          <CardContent>
            <p className="text-purple-200">{ancestryData.culture}</p>
          </CardContent>
        </Card>
      </div>
    </div>
  );
};

export default AncestryDetail;
