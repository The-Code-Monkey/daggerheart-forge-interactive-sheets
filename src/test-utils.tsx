import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

// Mock AuthContext
const mockAuthContext = {
  user: null,
  loading: false,
  signIn: jest.fn(),
  signOut: jest.fn(),
  signUp: jest.fn(),
};

const AuthProvider = ({ children }: { children: React.ReactNode }) => {
  return <div data-testid="auth-provider">{children}</div>;
};

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <BrowserRouter>
      <AuthProvider>{children}</AuthProvider>
    </BrowserRouter>
  );
};

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Export everything from testing-library/react
export * from "@testing-library/react";
export { customRender as render };

// Mock useAuth hook
export const mockUseAuth = () => mockAuthContext;
