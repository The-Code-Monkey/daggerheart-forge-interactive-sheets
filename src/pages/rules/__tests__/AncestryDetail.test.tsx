import { render, act } from "../../../test-utils";
import AncestryDetail from "../AncestryDetail";

// Mock react-router-dom
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({ ancestryName: "human" }),
  Link: ({ children, to }: { children: React.ReactNode; to: string }) => (
    <a href={to}>{children}</a>
  ),
}));

// Mock supabase helpers
jest.mock("@/integrations/supabase/helpers", () => ({
  getSingleAncestryBySlug: jest.fn().mockResolvedValue({
    id: 1,
    name: "Human",
    slug: "human",
    description: "Versatile and adaptable",
    features: [
      {
        name: "Test Feature",
        description: "Test description",
      },
    ],
  }),
}));

describe("AncestryDetail Page", () => {
  it("renders correctly", async () => {
    const { container } = render(<AncestryDetail />);
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 1000));
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
