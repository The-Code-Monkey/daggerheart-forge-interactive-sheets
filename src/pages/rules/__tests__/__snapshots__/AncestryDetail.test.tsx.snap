// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AncestryDetail Page renders correctly 1`] = `
<div
  data-testid="auth-provider"
>
  <div
    class="min-h-screen bg-nebula py-8 px-4"
  >
    <div
      class="max-w-4xl mx-auto"
    >
      <div
        class="mb-8"
      >
        <a
          href="/game-rules"
        >
          <button
            class="inline-flex items-center hover:cursor-pointer justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors duration-300 ease-in-out focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent h-10 px-4 py-2 text-purple-200 hover:text-white mb-4"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-arrow-left w-4 h-4 mr-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
            Back to Game Rules
          </button>
        </a>
        <div
          class="flex items-center gap-4 mb-4"
        >
          <div
            class="w-16 h-16 bg-linear-to-br from-yellow-400 to-orange-500 rounded-lg flex items-center justify-center"
          />
          <div>
            <h1
              class="text-4xl font-bold text-white"
            >
              Human
            </h1>
            <p
              class="text-xl text-purple-200"
            >
              Versatile and adaptable
            </p>
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border text-card-foreground shadow-sm bg-gradient-to-br bg-linear-to-br from-slate-800/80 to-slate-900/80 border-purple-500/30 backdrop-blur-xs col-span-2"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight text-white"
          >
            Ancestry Features
          </h3>
          <p
            class="text-sm text-purple-200"
          >
            Special abilities that define the 
            Human
             ancestry
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="space-y-4"
          >
            <div>
              <h3
                class="text-lg font-semibold text-white mb-2"
              >
                Test Feature
              </h3>
              <p
                class="text-purple-200"
              >
                Test description
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-6 mb-8"
      />
    </div>
  </div>
</div>
`;
